code/modules/mob/living/carbon/monkey/combat.dm

```diff
@@ -0,0 +1,433 @@
+			if(!target || !bodyDisposal || frustration >= 16)
+				back_to_idle()
+				return TRUE
+
+			if(target.pulledby != src && !istype(target.pulledby, /mob/living/carbon/monkey/))
+
+				walk2derpless(target.loc)
+
+				if(Adjacent(target) && isturf(target.loc))
+					a_intent = INTENT_GRAB
+					target.grabbedby(src)
+				else
+					var/turf/olddist = get_dist(src, target)
+					if((get_dist(src, target)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+
+			else if(!disposing_body)
+				walk2derpless(bodyDisposal.loc)
+
+				if(Adjacent(bodyDisposal))
+					disposing_body = TRUE
+					spawn(5)
+						bodyDisposal.stuff_mob_in(target, src)
+						disposing_body = FALSE
+						
+				else
+					var/turf/olddist = get_dist(src, bodyDisposal)
+					if((get_dist(src, bodyDisposal)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+
+			return TRUE
+
+
+
+	return IsStandingStill()
+
+/mob/living/carbon/monkey/proc/back_to_idle()
+	mode = MONKEY_IDLE
+	target = null
+	a_intent = INTENT_HELP
+	frustration = 0
+
+// handle de-aggro
+/mob/living/carbon/monkey/proc/attacked(mob/living/carbon/H)
+	if(!enemies[H])
+		return
+		
+	if(prob(25))
+		enemies[H] --
+
+	if(enemies[H] <= 0)
+		enemies.Remove(H)
+		if( target == H )
+			back_to_idle()
+
+/mob/living/carbon/monkey/proc/retaliate(mob/living/L)
+	mode = MONKEY_HUNT
+	target = L
+	enemies[L] += 4
+
+	if(a_intent != INTENT_HARM)
+		emote(pick("roar","screech"))
+		a_intent = INTENT_HARM
+
+/mob/living/carbon/monkey/attack_hand(mob/living/L)
+	if(L.a_intent == INTENT_HARM && prob(95))
+		retaliate(L)
+	else if(L.a_intent == INTENT_DISARM && prob(20))
+		retaliate(L)
+	return ..()
+
+/mob/living/carbon/monkey/attack_paw(mob/living/L)
+	if(L.a_intent == INTENT_HARM && prob(95))
+		retaliate(L)
+	else if(L.a_intent == INTENT_DISARM && prob(20))
+		retaliate(L)
+	return ..()
+
+/mob/living/carbon/monkey/attackby(obj/item/weapon/W, mob/user, params)
+	..()
+	if((W.force) && (!target) && (W.damtype != STAMINA) )
+		retaliate(user)
+
+/mob/living/carbon/monkey/bullet_act(obj/item/projectile/Proj)
+	if(istype(Proj ,/obj/item/projectile/beam)||istype(Proj,/obj/item/projectile/bullet))
+		if((Proj.damage_type == BURN) || (Proj.damage_type == BRUTE))
+			if(!Proj.nodamage && Proj.damage < src.health)
+				retaliate(Proj.firer)
+	..()
+
+/mob/living/carbon/monkey/hitby(atom/movable/AM, skipcatch = 0, hitpush = 1, blocked = 0)
+	if(istype(AM, /obj/item))
+		var/obj/item/I = AM
+		if(I.throwforce < src.health && I.thrownby && ishuman(I.thrownby))
+			var/mob/living/carbon/human/H = I.thrownby
+			retaliate(H)
+	..()
+
+/mob/living/carbon/monkey/Crossed(atom/movable/AM)
+	if(!IsDeadOrIncap() && ismob(AM) && target)
+		var/mob/living/carbon/monkey/M = AM
+		if(!istype(M) || !M)
+			return
+		knockOver(M)
+		return
+	..()
+
+/mob/living/carbon/monkey/proc/monkeyDrop(var/obj/item/A)
+	if(A)
+		unEquip(A)
+		A.loc = get_turf(src) // drop item works inconsistently
```

Comment:
This should never happen, and even if it does you should use A.forceMove(get_turf(src))